\documentclass[12pt]{article}

\usepackage{amssymb}
\usepackage{amsmath}
\usepackage{bm}
\usepackage{color}
\usepackage{float}
\usepackage{graphicx}
\usepackage{mathtools}
\usepackage[none]{hyphenat}
\usepackage{indentfirst}
\usepackage[labelfont=bf]{caption}
\usepackage[left=2.5cm,right=2.5cm, top=2.5cm,bottom=2.5cm]{geometry}
\usepackage{tcolorbox}
\usepackage{lineno}
\usepackage{setspace}
\DeclarePairedDelimiter\ceil{\lceil}{\rceil}

\usepackage[authoryear]{natbib}
\usepackage{hyperref}
\usepackage{yhmath}

\usepackage{dsfont}
\usepackage{mathtools}

\overfullrule 5pt

%\doublespacing

\usepackage{lipsum}

%\usepackage{adjustbox}
\usepackage{lscape}
\usepackage{afterpage}

\makeatletter
\renewcommand{\maketitle}{\bgroup\setlength{\parindent}{0pt}
\begin{flushleft}
\textbf{\@title}

  \@author
\end{flushleft}\egroup
}
\makeatother

\begin{document}

\linenumbers

\title{A Bayesian Model of the DNA Barcode Gap}

\author{Jarrett D. Phillips$^{1, 2*}$ (ORCID: 0000-0001-8390-386X)  \\
\textit{$^1$School of Computer Science, University of Guelph, Guelph, ON., Canada, N1G2W1} \\ \textit{$^2$Department of Integrative Biology, University of Guelph, Guelph, ON., Canada, N1G2W1} }

\date{}

\maketitle

\vspace{2mm}

\noindent \textbf{*Corresponding Author}: Jarrett D. Phillips$^{1}$

\noindent \textbf{Email Address}: jphill01@uoguelph.ca

\noindent \textbf{Running Title}: Bayesian inference for DNA Barcode Gap Estimation

\newpage

\begin{abstract}

\end{abstract}

\textbf{Keywords}: Bayesian inference, DNA barcoding, intraspecific genetic distance, \\ interspecific genetic distance, specimen identification, species discovery, Stan 

\vspace{2mm}

\section{Introduction}

Since its inception over 20 years ago, DNA barcoding \citep{hebert2003biological, hebert2003barcoding} has emerged as a robust method of specimen identification and species delimitation across myriad \\ taxonomic groups which have been sequenced at short, standardized gene regions like 5'-COI for animals. However, the success of the approach, particularly for regulatory and forensic applications, depends crucially on two important factors: (1) the availability of high-quality specimen records found in public reference sequence databases such as the Barcode of Life Data Systems (BOLD) \citep{ratnasingham2007bold}, and (2) the establishment of a DNA barcode gap --- the idea that the maximum genetic distance observed within species is much smaller than the minimum degree of marker variation found among species \citep{meyer2005dna, meier2008use}. Early work has demonstrated that the presence of a DNA barcode gap hinges strongly on extant levels of species haplotype diversity gauged from comprehensive specimen sampling at wide geographic and ecological scales \citep{bergsten2012effect}. Despite this, many taxa lack adequate separation in their pairwise intraspecific and interspecific genetic distances, thereby compromising rapid matching of unknown samples to expertly-validated references, and leading to cases of false positives (taxon oversplitting) and false negatives (excessive lumping of taxa) as a result of incomplete lineage sorting/retention of ancestral polymorphisms, hybridization/introgression, species synonymy, cryptic species diversity, and misidentifications \citep{hubert2015dna, phillips2022lack}.

Recent work has argued that DNA barcoding, in its current form, is lacking in statistical rigor, as most studies rely strongly on heuristic distance-based measures to infer taxonomic identity, and of these studies, few report measures of uncertainty, such as standard errors (SEs) and confidence intervals (CIs), around estimates of intraspecific and interspecific \\ variation, calling into question the existence of a true species' DNA barcode gap \citep{phillips2022lack}. To support this notion, novel nonparametric locus-specific metrics based on the multispecies coalescent \citep{rannala2003bayes, yang2017bayesian} were recently outlined and shown to hold strong promise when applied to predatory \textit{Agabus} (Coleoptera: Dytiscidae) diving beetles \citep{phillips2024measure}, a group, which, despite their morphological uniformity, is widely considered to be a taxonomic nightmare \citep{bergsten2012effect}. Further, the metrics indicate that sister species pairs from this taxon are often difficult to distinguish on the basis of their DNA barcode sequences \citep{phillips2024measure}. Using sequence data from three mitochondrial markers (5'-COI, 3'-COI, and CYTB) obtained from BOLD and GenBank, results highlight that DNA barcoding has been one-sided: there is a need to balance both the sufficient collection of specimens, as well as the deep sampling of species \citep{phillips2024measure}. DNA barcode libraries are biased toward the latter. The coalescent \citep{kingman1982coalescent} encompasses a backwards continuous-time stochastic Markov process of allelic sampling within natural, neutrally-evolving, species populations towards the Most Recent Common Ancestor (MRCA). The estimators represent a ``middle gound" between simple, yet arbitrary, heuristics such as the 2\% rule \citep{hebert2003biological} and the 10$\times$ rule \citep{hebert2004identification} and quantify the extent of asymmetric directionality of proportional genetic distance distribution overlap/separation for species within well-sampled taxonomic genera based on a straightforward distance count. The metrics can be employed in a variety of ways, including to assess performance of marker genes for species identication, as well as to assess whether computed values are consistent with population genetic-level parameters like effective population size ($N_e$), mutation rates ($\mu$) and divergence times ($\tau$) for species under study \citep{mather2019practical}. However, what appears to be missing is a rigorous statistical treatment of the DNA barcode gap, along an unbiased way to compute the statistical accuracy of the recommended estimators arising through problems inherent in frequentist maximum likelihood estimation for probability distributions having bounded positive support on the closed unit interval [0, 1]. To this end, here, a Bayesian model of the DNA barcode gap coalescent is introduced to rectify such issues. The model allows accurate estimation of posterior means, posterior standard deviations (SDs), posterior quantiles, and credible intervals (CrIs) for the metrics given datasets of intraspecific and interspecific genetic distances for species of interest.


\section{Methods}

\subsection{DNA Barcode Gap Metrics}

Recently, \citet{phillips2024measure} proposed novel nonparametric maximum likelihood \\ estimators (MLEs) of proportional overlap/separation between intraspecific and interspecific pairwise genetic distance distributions for a given species ($x$) to aid assessment of the DNA barcode gap as follows:

\begin{align}
p_x &= \frac{\#\{d_{ij} \geq a\}}{\#\{d_{ij}\}} \\[1mm]
q_x &= \frac{\#\{d_{XY} \leq b\}}{\#\{d_{XY}\}} \\[1mm]
p'_x &= \frac{\#\{d_{ij} \geq a'\}}{\#\{d_{ij}\}} \\[1mm]
q'_x &= \frac{\#\{d'_{XY} \leq b\}}{\#\{d'_{XY}\}}
\end{align}

\noindent where $d_{ij}$ are pairwise genetic distances within species, $d_{XY}$ are genetic distances among species for an entire genus of concern, and $d'_{XY}$ are combined interspecific distances for a target species and its closest neighbouring species. The notation \# reflects a count.  Quantities $a$, $a'$, and $b$ correspond to min($d_{XY}$), min($d'_{XY}$), and max($d_{ij}$), the minimum interspecific distance and the maximum intraspecific distance, respectively (\textbf{Figure 1}). \\ Hence, Equations (1)-(4) are simply empirical means of genetic distances falling at and below, or at and exceeding given distribution thresholds. Notice further that $a$/$a'$, and $b$ are also the first and $n$th order statistics, respectively. Equations (1)-(4) can be also be expressed in terms of empirical cumulative distribution functions (ECDFs) (see \textbf{Supplementary Information} for details). Genetic distances are easily computed from a model of DNA sequence evolution, such as uncorrected or corrected p-distances \citep{jukes1969evolution, kimura1980simple}. If a focal species is found to have multiple nearest neighbours, then the species possessing the smallest average pairwise interspecific distance is used. While these schemes differ \\ considerably from the usual definition of the DNA barcode gap laid out by \citet{meyer2005dna} and \citet{meier2008use}, they more accurately account for species' coalescence histories inferred from contemporaneous samples of DNA sequences, such as interspecific hybridization/introgression events \citep{phillips2024measure}. Note, genetic distances are \\ constrained to the unit interval [0, 1], whereas the metrics are defined only on [$a$/$a'$, b]. Values of the estimators obtained from equations (1)-(4) close to or equal to zero give evidence for separation between intraspecific and interspecific genetic distance distributions; that is, values suggest the presence of a DNA barcode gap for a target species. Conversely, values near or equal to one give evidence for distribution overlap; that is, values likely indicate the absence of a gap. 

\subsection{A Bayesian Implementation}

A major criticism of large sample (frequentist) theory is that it relies on asymptotic properties of the MLE (whose population parameter is assumed to be a fixed but unknown quantity), such as estimator normality and consistency. This problem is especially \\ pronounced in the case of binomial proportions. The estimated Wald standard error (SE) of the sample proportion, is given by 

\begin{equation}
\widehat{SE[\hat{p}}] = \sqrt{\frac{\hat{p}(1 - \hat{p})}{n}},
\end{equation}

\noindent where $\hat{p} = \frac{Y}{n}$ is the MLE, $Y$ is the number of successes ($Y = \sum_{i=1}^n{y_i}$) and $n$ is the number of trials (\textit{i.e.}, sample size). However, the above formula is problematic for several reasons. First, Equation (5) is a Normal approximation which makes use of the Central Limit Theorem (CLT); thus, large sample sizes are required for reliable estimation. When few observations are available, SEs will be large and inaccurate, leading to low statistical power to detect a true barcode gap when one sctually exists. Further, resulting interval estimates could span values less than zero or greater than one, or have zero width, which is practically meaningless. Second, when proprtions are exactly equal to zero or one, resulting SEs will be exactly zero, rendering Equation (5) completely useless. In the context of the proposed DNA barcode gap metrics, values obtained at the boundaries of their support are often encountered. Therefore, reliable calculation of SEs is not feasible. Given the importance of sufficient sampling of species genetic diversity for DNA barcoding initiatives, a different statistical estimation approach is necessary. Bayesian inference offers a natural path forward in this regard since it allows for straightforward specification of prior beliefs concerning unknown model parameters and permits the seamless propagation of uncertainty, when data is lacking and sample sizes are small, through integration with the likelihood function associated with true generating processes. As a consequence, because parameters are treated as random variables, Bayesian models are much more flexible and generally more easily interpretable compared to frequentist approaches, since, under the latter paradigm, entire posterior distributions, along with their summaries, are outputted, rather than just sampling distributions, p-values, and CIs as in the former case, thus allowing direct probability statements to be made. 

\subsection{The Model}

Essentially, from a statistical perspective, the goal herein is to nonparametrically estimate probabilities corresponding to extreme tail quantiles for positive highly skewed distributions on the unit interval  (or any closed subinterval thereof). Here, it is sought to numerically approximate the extent of proportional overlap/separation of intraspecific and interspecific pairwise genetic distance distributions within the subinterval [$a/a'$, $b$]. This is a challenging computational problem within the current study as detailed in subsequent sections. The usual approach employs Kernel Density Estimation (KDE), along with numerical or Monte Carlo integration; however, this requires careful selection of the bandwidth parameter, among other considerations. Here, for simplicity, a different route is taken. Counts, $y$, of overlapping genetic distances (as expressed in the numerator of Equations (1)-(4)) are treated as \\ binomially distributed with expectation $\mathbb{E}[Y] = k\theta$, where $k = \{N, C\}$ are total count vectors of intraspecific and combined genetic distances, respectively, for a target species along with its nearest neighbour species, and $k = M$ is a total count vector for all pairwise species comparisons. The quantity $\theta = \{p_x, q_x, p^{'}_x, q^{'}_x\}$. The metrics encompassing $\theta$ are presumed to follow a beta($\alpha$, $\beta$) distribution, with real shape parameters $\alpha$ and $\beta$, which is a natural choice of prior on probabilities. Such a scheme is quite convenient since the beta distribution is conjugate to the binomial distribution. Thus, the posterior distribution is also beta distributed. Parameters were given an uninformative Beta(1, 1) prior, which is equivalent to a standard uniform (Uniform(0, 1)) prior since it places equal probability on all parameter values within its support. As a result, the posterior is Beta($Y + 1$, $n - Y + 1$), from which various moments and other quantities, such as the expected value $\mathbb{E}[Y] = \frac{Y \hspace{0.5mm} + \hspace{0.5mm} 1}{n \hspace{0.5mm} + \hspace{0.5mm} 2}$ and variance $\mathbb{V}[Y] = \frac{(Y \hspace{0.5mm} + \hspace{0.5mm} 1)(n \hspace{0.5mm} - Y \hspace{0.5mm} + \hspace{0.5mm} 1)}{(n \hspace{0.5mm} + \hspace{0.5mm} 2)^2(n \hspace{0.5mm} + \hspace{0.5mm} 3)}$, can be easily calculated. In general however, when possible, it is always advisable to incorporate prior information, even if only weak, rather than simply imposing complete ignorance in the form of a flat prior distribution. With sufficient data, the choice of prior distribution becomes less important since the posterior will be dominated by the likelihood.  The full univariate Bayesian model for species $x$ is thus given by

\begin{align}
\notag y_\mathrm{lwr} &\sim \mathrm{Binomial}(N, p_\mathrm{lwr}) \\ 
\notag y_\mathrm{upr} &\sim \mathrm{Binomial}(M, p_\mathrm{upr}) \\ 
y^{'}_\mathrm{lwr} &\sim \mathrm{Binomial}(N, p^{'}_\mathrm{lwr}) \\ 
 \notag y^{'}_\mathrm{upr} &\sim \mathrm{Binomial}(C, p^{'}_\mathrm{upr}) \\ 
\notag p_\mathrm{lwr}, p_\mathrm{upr}, p^{'}_\mathrm{lwr}, p^{'}_\mathrm{upr}
&\sim \mathrm{Beta}(1, 1).
\end{align}

The model, which is inherently vectorized to allow processing of multiple species datasets simultaneously, was fitted using the Stan probabilistic programming language  \citep{carpenter2017stan} framework for Hamiltonian Monte Carlo (HMC) via the No-U-Turn Sampler (NUTS) sampling algorithm \citep{hoffman2014no} through the {\tt rstan} R package \citep{stan2023rstan}. Four Markov chains were run for 2000 iterations each in parallel across four cores with random parameter initializations. Within each chain, a total of 1000 samples was discarded as warmup (\textit{i.e.}, burnin) to reduce dependence on starting conditions. Further, 1000 post-warmup draws were utilized per chain. Because HMC/NUTS results in dependent samples that are minimally autocorrelated, chain thinning is not required. Each of these reflect default Markov Chain Monte Carlo (MCMC) settings in Stan. Since the DNA barcode gap metrics often attain values very close to zero (suggesting no overlap and complete separation) and/or very near one (indicating no separation and complete overlap), in addition to more intermediate values, a noninformative Beta($\frac{1}{2}, \frac{1}{2}$) prior, which is U-shaped symmetric and places greater probability density at the extremes of the distribution due to its heavier tails, while still allowing for variability in parameter estimates within intermediate values along its domain, was also attempted. However, this resulted in several divergent transitions, among other pathologies, imposed by complex geometry (\textit{i.e.}, curvature) in the posterior space, despite remedies to resolve them, such as lowering the step size of the HMC/NUTS sampler. Note that this prior is Jeffreys' prior, which is proportional to the square root of the Fisher information and has several desirable statistical properties, most notably invariance to reparameterization. 

To validate the overall correctness of the proposed statistical model given by \\ Equation (6), in addition to generating MLEs as a means of comparison, posterior predictive checks were also employed to generate binomial random variates in the form of counts from the posterior predictive distribution; that is $\gamma = \{Np_x, Mq_x, Np^{'}_x, Cq^{'}_x\}$ to verify that the model adequately captures relevant features of the observed data.

The proposed Bayesian model outlined herein has a straightforward interpretation \\ (\textbf{Table 1}). 

\section{Case Study}

\section{Discussion}


\newpage

\section*{Supplementary Information}

Information accompanying this article can be found in Supplemental Information.pdf.

\section*{Data Availability Statement}

Raw data, R, and Stan code can be found on GitHub at: \\ https://github.com/jphill01/Bayesian-DNA-Barcode-Gap-Coalescent.

\section*{Acknowledgements}

We wish to recognise the valuable comments and discussions of Daniel (Dan) Gillis, Robert (Bob) Hanner, Robert (Rob) Young, and XXX anonymous reviewers.

We acknowledge that the University of Guelph resides on the ancestral lands of the Attawandaron people and the treaty lands and territory of the Mississaugas of the Credit. We recognize the significance of the Dish with One Spoon Covenant to this land and offer our respect to our Anishinaabe, Haudenosaunee and M{\'e}tis neighbours as we strive to strengthen our relationships with them.

\section*{Funding}

None declared.

\section*{Conflict of Interest}

None declared.

\section*{Author Contributions}

JDP wrote the manuscript, wrote R and Stan code, approved all developed code as well as analysed and interpreted all experimental results. 

\bibliographystyle{humannat}
\bibliography{References}

\section*{Figures and Tables}

\begin{figure}[H]

\centering

\includegraphics[width=1.0\textwidth]{Figure 1}

\caption{Modified depiction from \citet{meyer2005dna} and \citet{phillips2024measure} of the overlap/separation of pairwise intraspecific and interspecific genetic distances ($\delta$) for calculation of the DNA barcode gap metrics ($p_x$ and $q_x$) for a hypothetical species $x$. The minimum interspecific distance is denoted by $a$ and the maximum intraspecific distance is indicated by $b$. The quantity $f(\delta)$ is akin to a kernel density estimate of the probability density function of pairwise genetic distances. A similar visualization can be displayed for $p^{'}_x$ and $q^{'}_x$ within the interval [$a'$, $b$].}

\end{figure}



\begin{table}[htbp]
    \centering
    \small
    \caption{Interpretation of the model within [$a/a'$, $b$]}
    \label{tab:parameters}
    \begin{tabular}{cp{0.7\linewidth}}
    \hline
    \textbf{Parameter} & \textbf{Explanation} \\
    \hline
    $p_{\text{lwr}}$ & When $p_{\text{lwr}}$ is close to 0 (1), it suggests that the probability of intraspecific (interspecific) distances being larger (smaller) than interspecific (intraspecific) distances is low (high) on average, while the probability of interspecific (intraspecific) distances being larger (smaller)  than intraspecific (interspecific) distances is high (low) on average; that is, there is (no) evidence for a DNA barcode gap.\\
        & \\[-2mm]
    $p_{\text{upr}}$ & When $p_{\text{upr}}$ is close to 0 (1), it suggests that the probability of interspecific (intraspecific) distances being larger (smaller) than intraspecific (interspecific) distances is high (low) on average, while the probability of intraspecific (interspecific) distances being larger (smaaler) than interspecific (intraspecfic) distances is low (high) on average; that is, there is (no) evidence for a DNA barcode gap. \\
        & \\[-2mm]
    $p^{'}_{\text{lwr}}$ & When $p^{'}_{\text{lwr}}$ is close to 0 (1), it suggests that the probability of intraspecific (combined interspecific distances for a target species and its nearest neighbour species) distances being larger than combined interspecific distances for a target species and its nearest neighbour species (intraspecific distances) is low (high) on average, while the probability of combined interspecific distances for a target species and its nearest neighbour species (intraspecfic distances) being larger than intraspecific distances (combined interspecific distances for a target species and its nearest neighbour species) is high (low) on average; that is, there is (no) evidence for a DNA barcode gap.\\
        & \\[-2mm]
    $p^{'}_{\text{upr}}$ & When $p^{'}_{\text{upr}}$ is close to 0 (1), it suggests that the probability of combined interspecific distances for a target species and its nearest neighbour species (intraspecific distances) being larger than intraspecific distances (combined interspecific distances for a target species and its nearest neighbour species) is high (low) on average, while the probability of intraspecific distances (combined interspecific distances for a target species and its nearest neighbour species) being larger than combined interspecific distances for a target species and its nearest neighbour species (intraspecific distances) is low (high) on average; that is, there is (no) evidence for a DNA barcode gap.\\
    \hline
    \end{tabular}
\end{table}



\end{document}